core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -137,6 +138,14 @@ class DAGScheduler(
   private[scheduler] val eventProcessLoop = new DAGSchedulerEventProcessLoop(this)
   taskScheduler.setDAGScheduler(this)
 
+  // Number of map, reduce tasks above which we do not assign preferred locations
```

Comment:
Could you add a comment here saying that we limit the size because of scalability issues with sorting the best locations?